doc/rds-day-2.md

```diff
@@ -0,0 +1,184 @@
+
+There are 188 static parameters in `mysql8.0` family, and similar number of them are in other parameter group families as well. You can get a list of static parameters by using the `aws rds describe-engine-default-parameters` command. 
+
+These fields and parameters need to be stored for use by whatever check mechanism you choose, and they need to be updated regularly.
+
+It is also worth pointing out that when a user updates a parameter in a parameter group, the changes to parameter groups themselves usually work without problems. However, it is often not the intention of these changes. The intention of changes is to change the parameter and apply it to a running instance. In both providers, changes to static parameters are not actually applied until the next maintenance window or a manual restart is issued. 
+
+We will discuss a few approaches to this problem below. **Whichever approach you choose, it is important for the check mechanisms to work reliably. It's easy to lose users' trust when checks say there will be a restart but no restart happened. Or worse, checks did not detect potential restarts and caused an outage.**
+
+### Check during PR
+Use Pull Request as a checkpoint and ensure developers are aware of potential consequences of the changes. An example process may look something like the following. 
+
+```mermaid
+flowchart TD
+    Comment(Comment on PR)
+
+    subgraph Workflow
+        GetChangedFiles(Get changed files)
+        GetChangedFiles(Get changed files)
+        StepCheck(Will this cause a restart?)
+    end
+
+    subgraph Data Source 
+        FieldDefinitions(Fields that need restarting)
+    end 
+
+    Restart(Restart immediately)
+
+    FieldDefinitions <--reference--> StepCheck
+
+    PR(PR Created) --trigger--> GetChangedFiles --> StepCheck --yes--> Comment --> Approval(Wait for Approval) --> Merge --> GitOps(GitOps Tooling)
+    StepCheck--no--> Approval
+    GitOps --apply changes now --> Restart --> Done
+    GitOps --wait until next \n maintenance window--> Done
+```
+
+In this example, whenever a pull request is created, a workflow is executed and a comment is created on the PR warning the developers of potential impacts. When developers approve the PR, it implies that they are aware of consequences.
+To check if a PR is impacted, you can use of the following options:
+- Parse git diff and search for changes to "dangerous" fields
+- Use `kubectl diff` then look for changes to "dangerous" fields. This requires read access to the target cluster but more accurate.
+
+
+### Check at runtime
+
+Another approach is to deny such operation at runtime using a policy engine and/or custom validating web hook unless certain conditions are met. This means problems with RDS configuration is communicated to the developers through their GitOps tooling by providing reasons for denial.
```

Comment:
IMO This should be in-place, no matter what. Extra validation will for sure be beneficial in one way or another.

Reply:
Agreed. I will update the doc.